presto-main/src/main/java/com/facebook/presto/sql/planner/LiteralInterpreter.java

```diff
@@ -190,6 +192,10 @@ else if (value.equals(Float.POSITIVE_INFINITY)) {
             return new BooleanLiteral(object.toString());
         }
 
+        if (type.equals(DATE)) {
```

Comment:
test?

Reply:
Currently there are none direct tests of `LiteralInterpreter`. The effect of applying this change will be only visible in CBO where we do better stats math thanks to this.